browser/net/ipfs_redirect_network_delegate_helper.cc

```diff
@@ -48,8 +51,8 @@ int OnHeadersReceived_IPFSRedirectWork(
     GURL* allowed_unsafe_redirect_url,
     const brave::ResponseCallback& next_callback,
     std::shared_ptr<brave::BraveRequestInfo> ctx) {
-  if (!ctx->browser_context ||
-      IsIpfsResolveMethodDisabled(ctx->browser_context)) {
+  auto* prefs = user_prefs::UserPrefs::Get(ctx->browser_context);
+  if (!ctx->browser_context || IsIpfsResolveMethodDisabled(prefs)) {
```

Comment:
we'd better check ctx->browser_context before doing UserPrefs::Get(ctx->browser_context)